src/test/java/org/wildfly/extension/elytron/DomainTestCase.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * JBoss, Home of Professional Open Source
+ *
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.elytron;
+
+import mockit.Mock;
+import mockit.MockUp;
+import mockit.integration.junit4.JMockit;
+import org.jboss.as.subsystem.test.AbstractSubsystemTest;
+import org.jboss.as.subsystem.test.KernelServices;
+import org.jboss.msc.service.ServiceName;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.security.auth.principal.NamePrincipal;
+import org.wildfly.security.auth.server.MechanismConfiguration;
+import org.wildfly.security.auth.server.MechanismRealmConfiguration;
+import org.wildfly.security.auth.server.SecurityDomain;
+import org.wildfly.security.auth.server.SecurityIdentity;
+import org.wildfly.security.auth.server.ServerAuthenticationContext;
+import org.wildfly.security.authz.PermissionMapper;
+import org.wildfly.security.authz.Roles;
+import org.wildfly.security.permission.PermissionVerifier;
+
+import javax.security.auth.x500.X500Principal;
+import java.io.FilePermission;
+import java.security.Principal;
+
+
+/**
+ * @author <a href="mailto:jkalina@redhat.com">Jan Kalina</a>
+ */
+@RunWith(JMockit.class)
+public class DomainTestCase extends AbstractSubsystemTest {
+
+    public DomainTestCase() {
+        super(ElytronExtension.SUBSYSTEM_NAME, new ElytronExtension());
+    }
+
+    private KernelServices services = null;
+
+    private String filesystemFile = RealmsTestCase.class.getResource("/org/wildfly/extension/elytron/filesystem-realm").getFile();
+    private String propertiesFile = RealmsTestCase.class.getResource("/org/wildfly/extension/elytron/testingrealm1-users.properties").getFile();
+
+    private void init() throws Exception {
+        new ClassLoadingAttributeDefinitionsMock(); // mock classloader obtaining
+        String subsystemXml = "<subsystem xmlns=\"" + ElytronExtension.NAMESPACE + "\">\n" +
+                "    <security-domains>\n" +
```

Comment:
I think I would prefer to keep the XML in an XML file that is read in by the test.  That way when working with IDEs we edit source code as source code and XML within an XML editor.